README.md

```diff
@@ -9,20 +9,88 @@
 
 ## Overview
 
-In this lesson we're going implement a simple `<Counter />` component. We're
-going to focus on the stores and actions while building this application.
+In this lesson we're going implement a simple counter application.
+Our focus will be on the stores and actions, how they interact with each other
+and how we can wire them up to components.
 
-There are two stores: `Store` and `CounterStore`. `CounterStore` inherits from
-`Store`.
+Our counter has two buttons used for incrementing and decrementing a number.
+This is how it's going to look like:
 
-The global application state that we're updating is a simple counter,
-represented by a number.
+![Screenshot](assets/screenshot.png)
+
+## Directory Structure
+
+```
+components/
+└── App.js
+stores/
+├── Store.js
+└── counterStore.js
+actions
+└── index.js
+```
+
+Our directory structure looks rather boring this time, since we're mostly going
+to focus on the global store of our application.
+
+## Components
+
+Our application has a single component called `<App />`. `<App />` needs to be
+"wired" up to the `counterStore` using the component's lifecycle methods:
+
+* `componentDidMount` is going to add an event listener to the `counterStore`.
+    If the store is being updated, the state of our `<App />` component should
+    be updated as well.
+
+* `componentWillUnmount` is going to remove the event listener.
 
 Our `<App />` component renders two buttons and the actual counter number. If
 we click on `+`, the counter will be incremented by `1`. If we click `-`, the
 counter will be decremented by `1`.
 
-There are two actions: `increment()` and `decrement()`.
+## Stores
+
+There are two stores: `Store` and `CounterStore`. `CounterStore` inherits from
+`Store`. `counterStore` exports a single instance of `CounterStore`, whereas
+`Store` exports a class that we can inherit from in the future.
+
+We first need to implement the `Store`'s instance methods before we can
+implement the `CounterStore` class.
+
+The global application state that we're updating is a simple counter,
+represented by a number. The initial count will be set to `0`, which will also
+be the initial state of the `counterStore`.
+
+### Actions
+
+There are two actions: `increment()` and `decrement()`. The action creators are
+directly wired up to the global `counterStore` singleton.
+
+`increment()` and `decrement()` respectively update the `counterStore` by
+
+1. retrieving the previous state using `store.getState()`.
+2. adding / subtracting `1` from the previous state and calling
+    `store.setState(newState)`
+
+### Stores -> Components
+
+Our `<App />` component doesn't directly update the store, but subscribes to
+its updates. The store is the single source of truth of our application, but is
+completely decoupled from individual components. In other words, there is no way
+how a component can directly update a store.
```

Comment:
'....there is no way a component can...' (remove the _how_)

*or*

'...there is no way for a component to directly update the store'